components/debug/EnhancedDebugPanel.tsx

```
@@ -0,0 +1,583 @@
+    if (selectedLevel !== 'ALL') {
+      filtered = filtered.filter(log => log.level === selectedLevel);
+    }
+    
+    return filtered.slice(-100); // Show last 100 logs for performance
+  }, [logs, searchTerm, fuzzySearch, selectedCategory, selectedLevel, searchLogs]);
+
+  // Group logs by different criteria
+  const groupedLogs = useMemo(() => {
+    switch (viewMode) {
+      case 'flow':
+        const flowGroups = new Map<string, typeof filteredLogs>();
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)